kernel/src/scan/mod.rs

```diff
@@ -211,40 +245,19 @@ impl Scan {
                     0
                 };
 
-                let read_result = if partition_fields.is_empty() {
-                    read_result
-                } else {
-                    debug!("Adding back in partition columns");
-                    let final_schema = self.schema();
-                    let mut fields = vec![None; final_schema.fields.len()];
-                    for select_field in read_schema.fields() {
-                        let field_position = final_schema
-                            .field_index(&select_field.name)
-                            .ok_or_else(|| Error::missing_column(&select_field.name))?;
-                        fields[field_position] = Some(Expression::column(select_field.name()));
-                    }
-                    for field in &partition_fields {
-                        let value_expression = parse_partition_value(
-                            add.partition_values.get(field.name()),
-                            field.data_type(),
-                        )?;
-                        let field_position = final_schema
-                            .field_index(&field.name)
-                            .ok_or_else(|| Error::missing_column(&field.name))?;
-                        fields[field_position] = Some(Expression::Literal(value_expression));
-                    }
-                    debug!("Final fields to evaluate for partition columns: {fields:?}");
-                    let fields = fields
-                        .into_iter()
-                        .map(|opt| opt.ok_or_else(|| Error::generic("Invalid columns in select")))
-                        .try_collect()?;
+                let read_result = if have_partition_cols {
```

Comment:
Since we have a nice `Option` available, shouldn't this be something like:
```rust
let read_result = match read_expression {
  Some(read_expression) => engine_interface
    .get_expression_handler()
    .get_evaluator(...)
    .evaluate(...),
  None => read_result,
}
```
(avoids the unwrap below)